### PR TITLE
[6.x] Scripts/Monk: Renewing Mist

### PR DESCRIPTION
**Pull request by P-Kito**
**Changes proposed**:
- Renewing Mist jumps from target to target and applies the heal
- Sniffs: http://npaste.de/p/GJbK/

**Target branch(es)**: 6x

**Tests performed**: See TODO list

**Known issues and TODO list**:
- [x] Test Ingame
- [ ] Code Improvements (!!)
- [ ] Bug: Aura is not "Reapplied" when the monk casts it again on a target which has it already (and thus basepoints are not set back to original value of 3.

### CODE REVIEW

**Comment by P-Kito** [comment]
Will re-do this script today - I sniffed it today.
http://npaste.de/p/GJbK/

File: src/server/scripts/Spells/spell_monk.cpp [hunk 1]
```diff
@@ -34,6 +34,10 @@ enum MonkSpells
     SPELL_MONK_CRACKLING_JADE_LIGHTNING_KNOCKBACK_CD    = 117953,
     SPELL_MONK_PROVOKE_SINGLE_TARGET                    = 116189,
     SPELL_MONK_PROVOKE_AOE                              = 118635,
+    SPELL_MONK_RENEWING_MIST                            = 115151,
+    SPELL_MONK_RENEWING_MIST_PERIODIC_HEAL              = 119611,
+    SPELL_MONK_RENEWING_MIST_TARGET_SELECT              = 119607,
+    SPELL_MONK_RENEWING_MIST_DUMMY_VISUAL               = 119647,
     SPELL_MONK_STANCE_OF_THE_SPIRITED_CRANE             = 154436,
```
Comment (GitVerge) [inline src/server/scripts/Spells/spell_monk.cpp]: the final "," again xD

Comment (P-Kito) [inline src/server/scripts/Spells/spell_monk.cpp]: Who cares about that XD

File: src/server/scripts/Spells/spell_monk.cpp [hunk 2]
```diff
@@ -186,9 +190,172 @@ class spell_monk_provoke : public SpellScriptLoader
...
+            }
+
+            void FilterTargets(std::list<WorldObject*>& targets)
+            {
+                if (SpellEffectInfo const* eff = GetSpellInfo()->GetEffect(EFFECT_1))
+                {
+                    targets.remove(GetExplTargetWorldObject());
+                    if (targets.empty())
+                    {
+                        emptyTargets = true;
```
Comment (GitVerge) [inline src/server/scripts/Spells/spell_monk.cpp]: Is that bool really needed? OnEffectHitTarget is meant to be executed after hitting someone from targets, so I assume it won't be null in HandleDummy. Tell me if you tested without this check and what happened just to know for the next time.

Comment (P-Kito) [inline src/server/scripts/Spells/spell_monk.cpp]: I think you are right.

File: src/server/scripts/Spells/spell_monk.cpp [hunk 3]
```diff
@@ -186,9 +190,172 @@ class spell_monk_provoke : public SpellScriptLoader
...
+                        return;
+                    }
+                    targets.sort(Trinity::HealthPctOrderPred());
+                    targets.resize(eff->BasePoints);
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_monk.cpp]: never use baseponts directly

**Pull request closed without merging**